### PR TITLE
Fix the issues in nested types, array types and empty messages in protobuff tool 

### PR DESCRIPTION
**Pull request by Yasara123**

### CODE REVIEW

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/builder/components/StubObject.java [hunk 1]
```diff
@@ -43,6 +45,17 @@ public void addStub(String stubTypeName, String stubType) {
...
+            blockingFunctionsObj.setInputComma(null);
+            blockingFunctionsObj.setInputAttributeName(null);
+            blockingFunctionsObj.setInputDataType(null);
+            blockingFunctionsObj.initEmptyStruct();
+        }
+        if(EMPTY_DATATYPE_NAME.equals(outputDataType)) {
+            blockingFunctionsObj.setOutputComma(null);
+            blockingFunctionsObj.setOutputDataType(null);
+            blockingFunctionsObj.ignoreCast();
+        }
```
Comment (daneshk) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/builder/components/StubObject.java]: Need space in if clause. line 48 and 54

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/builder/components/BlockingFunction.java [hunk 2]
```diff
@@ -85,4 +107,76 @@ public String getConnectorId() {
...
+    
+    public String getInputComma() {
+        return inputComma;
+    }
+    
+    public void setInputComma(String inputComma) {
+        this.inputComma = inputComma;
+        if(inputComma == null){
+            this.space = null;
+        }
```
Comment (daneshk) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/builder/components/BlockingFunction.java]: checkstyle issue at line 117

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/MessageConstants.java [hunk 3]
```diff
@@ -122,5 +122,16 @@
     public static final String BOOLEAN = "boolean";
     public static final String BLOB = "blob";
     
-    
+    //stub template builder constants
+    public static final String COMMA = ",";
+    public static final String INPUT_TYPE_NAME = "req";
+    public static final String RESULT_TYPE_NAME = "result";
+    public static final String IGNORE_CAST = "_";
+    public static final String INIT_EMPTY_STUB = "Empty req = {};";
```
Comment (daneshk) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/MessageConstants.java]: can we move this code to template

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/builder/components/StubObject.java (hunk 1 above)
Comment (Yasara123) [reply stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/builder/components/StubObject.java]: Fixed in e2b8379

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/builder/components/BlockingFunction.java (hunk 2 above)
Comment (Yasara123) [reply stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/builder/components/BlockingFunction.java]: Fixed in e2b8379

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/MessageConstants.java [hunk 4]
```diff
@@ -122,5 +122,16 @@
...
+    public static final String EMPTY_DATATYPE_NAME = "Empty";
+    public static final String DIAMOND_CAST_OPEN = "<";
+    public static final String DIAMOND_CAST_CLOSE = ">";
+    public static final String BRACKET_OPEN = "(";
+    public static final String BRACKET_CLOSE = ")";
+    public static final String SPACE = " ";
```
[the remark this answers is not in the document]
Comment (Yasara123) [reply stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/MessageConstants.java]: Using Template we cannot handle if-else itself. So we have to handle them in java level and cannot move that to there.

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/MessageConstants.java (hunk 3 above)
Comment (Yasara123) [reply stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/MessageConstants.java]: Using Template we cannot handle if-else itself. So we have to handle them in java level and cannot move that to there.

**Pull request merged**